test/test_http_server.c

```diff
@@ -128,6 +130,9 @@ static int my_post_handler(mtev_http_rest_closure_t *restc, int npats, char **pa
   return EVENTER_READ | EVENTER_WRITE | EVENTER_EXCEPTION;
 }
 static int my_safe_handler(mtev_http_rest_closure_t *restc, int npats, char **pats) {
+  UNUSED(npats);
```

Comment:
I would rather adopt `C23` than add this everywhere.  Was there an error compiling this?

Reply:
We can mark this as TODO once we have compiler support for C23. The issue was unused-variable warnings.

Reply:
I know, but in C23 you can do

```c
static int my_safe_handler(mtev_http_rest_closure_t *restc, int /*npats*/, char ** /*pats*/) {
```

or 

```c
static int my_safe_handler(mtev_http_rest_closure_t *restc, int , char **) {
```

or

```c
static int my_safe_handler(mtev_http_rest_closure_t *restc, [[maybe_unused]] int npats, [[maybe_unused]] char **pats) {
```

which matches C++.  I would rather do it the C way until then which is to say disable that warning for arguments than have this stuff everywhere instead of adding ``TODO`` everywhere

Reply:
Decided to leave the unused-parameter warnings (not unused-variable, I misspoke above) but make them not errors. When we have C23 we can decorate them appropriately to quash the warning.